plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-products-controller.php

```diff
@@ -70,6 +72,67 @@ protected function get_images( $product ) {
 		return $images;
 	}
 
+	/**
+	 * Get a collection of products and add other search criteria to WP_Query.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function get_items( $request ) {
+		// Add filters for search criteria in product postmeta via the lookup table.
+		$can_search_products_by_sku = $this->can_search_products_by_sku( $request );
+		if ( $can_search_products_by_sku ) {
+			add_filter( 'posts_where', array( __CLASS__, 'add_search_criteria_to_wp_query_filter' ), 10, 2 );
+			add_filter( 'posts_join', array( __CLASS__, 'add_search_criteria_to_wp_query_join' ), 10, 2 );
+		}
+
+		$response = parent::get_items( $request );
+
+		// Remove filters for search criteria in product postmeta via the lookup table.
+		if ( $can_search_products_by_sku ) {
+			remove_filter( 'posts_where', array( __CLASS__, 'add_search_criteria_to_wp_query_filter' ), 10 );
+			remove_filter( 'posts_join', array( __CLASS__, 'add_search_criteria_to_wp_query_join' ), 10 );
+		}
+		return $response;
+	}
+
+	/**
+	 * Add in conditional search filters for products.
+	 *
+	 * @param string $where Where clause used to search posts.
+	 * @param object $wp_query WP_Query object.
+	 * @return string
+	 */
+	public static function add_search_criteria_to_wp_query_filter( $where, $wp_query ) {
+		global $wpdb;
+		$search = $wp_query->get( self::$wp_query_search_param );
+		if ( ! empty( $search ) ) {
+			$like_search = '%' . $wpdb->esc_like( $search ) . '%';
+			$conditions  = array(
+				$wpdb->prepare( 'wc_product_meta_lookup.sku LIKE %s', $like_search ),
+			);
+			$where      .= ' OR (' . implode( ' OR ', $conditions ) . ')';
+		}
+		return $where;
+	}
+
+	/**
+	 * Join `wc_product_meta_lookup` table when product search query is present.
+	 *
+	 * @param string $join Join clause used to search posts.
+	 * @param object $wp_query WP_Query object.
+	 * @return string
+	 */
+	public static function add_search_criteria_to_wp_query_join( $join, $wp_query ) {
+		global $wpdb;
+		$search = $wp_query->get( self::$wp_query_search_param );
+		if ( ! empty( $search ) ) {
+			$join .= " LEFT JOIN {$wpdb->wc_product_meta_lookup} wc_product_meta_lookup
```

Comment:
Should this just be `LEFT JOIN {$wpdb->wc_product_meta_lookup}`?

Reply:
I suspect this is copied from [here](https://github.com/woocommerce/woocommerce/blob/cfa828d70d39b2659d3621ce487b1022cd0253f0/plugins/woocommerce/includes/class-wc-query.php#L721-L734), which also specifies the alias. Probably not necessary, but then on the next line we'd need to use `$wpdb->wc_product_meta_lookup.product_id`.

Could that method be changed to public? Then we could avoid duplicating its code here.

Reply:
> Could that method be changed to public?

Either that, or perhaps this logic should get added in the WC_Query class instead of this REST controller...